validator/client/src/main/java/tech/pegasys/teku/validator/client/signer/ExternalSigner.java

```diff
@@ -0,0 +1,162 @@
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.validator.client.signer;
+
+import static java.util.Collections.emptyMap;
+import static tech.pegasys.teku.core.signatures.SigningRootUtil.signingRootForRandaoReveal;
+import static tech.pegasys.teku.core.signatures.SigningRootUtil.signingRootForSignAggregateAndProof;
+import static tech.pegasys.teku.core.signatures.SigningRootUtil.signingRootForSignAggregationSlot;
+import static tech.pegasys.teku.core.signatures.SigningRootUtil.signingRootForSignAttestationData;
+import static tech.pegasys.teku.core.signatures.SigningRootUtil.signingRootForSignBlock;
+import static tech.pegasys.teku.core.signatures.SigningRootUtil.signingRootForSignVoluntaryExit;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import java.net.URI;
+import java.net.URL;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpRequest.BodyPublishers;
+import java.net.http.HttpResponse;
+import java.net.http.HttpResponse.BodyHandlers;
+import java.time.Duration;
+import java.util.Map;
+import org.apache.tuweni.bytes.Bytes;
+import tech.pegasys.teku.bls.BLSPublicKey;
+import tech.pegasys.teku.bls.BLSSignature;
+import tech.pegasys.teku.core.signatures.Signer;
+import tech.pegasys.teku.datastructures.blocks.BeaconBlock;
+import tech.pegasys.teku.datastructures.operations.AggregateAndProof;
+import tech.pegasys.teku.datastructures.operations.AttestationData;
+import tech.pegasys.teku.datastructures.operations.VoluntaryExit;
+import tech.pegasys.teku.datastructures.state.ForkInfo;
+import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.provider.JsonProvider;
+
+public class ExternalSigner implements Signer {
+  public static final String EXTERNAL_SIGNER_ENDPOINT = "/api/v1/eth2/sign";
+  private final JsonProvider jsonProvider = new JsonProvider();
+  private final URL signingServiceUrl;
+  private final BLSPublicKey blsPublicKey;
+  private final Duration timeout;
+  private final HttpClient httpClient = HttpClient.newHttpClient();
+
+  public ExternalSigner(
+      final URL signingServiceUrl, final BLSPublicKey blsPublicKey, final Duration timeout) {
+    this.signingServiceUrl = signingServiceUrl;
+    this.blsPublicKey = blsPublicKey;
+    this.timeout = timeout;
+  }
+
+  @Override
+  public SafeFuture<BLSSignature> createRandaoReveal(final UInt64 epoch, final ForkInfo forkInfo) {
+    return sign(signingRootForRandaoReveal(epoch, forkInfo), emptyMap());
```

Comment:
can we include the type on all the sign requests. although there will be no additional data it would be nice for our metrics to seperate the type of signing request.

Reply:
will do.

Reply:
done